File: frontend/src/api/project.ts
import api from './index'

export interface Project {
  id: number
  name: string
  description: string
  status: string
  start_date?: string
  end_date?: string
  task_count: number
  completed_count: number
  created_at: string
  updated_at?: string
}

export interface ProjectListResponse {
  items: Project[]
  total: number
}

export const projectApi = {
  list(params?: { status?: string; page?: number; size?: number }) {
    return api.get<any, ProjectListResponse>('/api/projects', { params })
  },
  getAll() {
    return api.get<any, Project[]>('/api/projects', { params: { size: 999 } })
  },
  get(id: number) {
    return api.get<any, Project>(`/api/projects/${id}`)
  },
  create(data: { name: string; description?: string; status?: string; start_date?: string; end_date?: string }) {
    return api.post<any, Project>('/api/projects', data)
  },
  update(id: number, data: Partial<Project>) {
    return api.put<any, Project>(`/api/projects/${id}`, data)
  },
  delete(id: number) {
    return api.delete(`/api/projects/${id}`)
  }
}

File: backend/app/schemas/project.py
"""Project schemas."""

from pydantic import BaseModel
from datetime import date, datetime
from typing import Optional


class ProjectCreate(BaseModel):
    name: str
    description: str = ""
    start_date: Optional[date] = None
    end_date: Optional[date] = None


class ProjectUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    status: Optional[str] = None
    start_date: Optional[date] = None
    end_date: Optional[date] = None


class ProjectResponse(BaseModel):
    id: int
    name: str
    description: str
    status: str
    start_date: Optional[date]
    end_date: Optional[date]
    task_count: int = 0  # 关联任务数量
    completed_count: int = 0  # 已完成任务数量
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
